Sources/MapboxNavigation/NavigationViewportDataSource.swift

```diff
@@ -174,106 +166,154 @@ public class NavigationViewportDataSource: ViewportDataSource {
             
-            var center: CLLocationCoordinate2D = location.coordinate
-            if let adjustedCenter = centerLineString.coordinateFromStart(distance: centerCoordDistance) {
-                center = adjustedCenter
+            if options.followingCameraOptions.centerUpdatesAllowed {
+                var center: CLLocationCoordinate2D = location.coordinate
+                let centerLineString = LineString([
+                    location.coordinate,
+                    (coordinatesToManeuver + coordinatesForManeuverFraming)
+                        .map({ mapView.point(for: $0) }).boundingBoxPoints
+                        .map({ mapView.coordinate(for: $0) }).centerCoordinate
+                ])
+                let centerLineStringTotalDistance = centerLineString.distance() ?? 0.0
+                let centerCoordDistance = centerLineStringTotalDistance * (1 - pitchСoefficient)
+                if let adjustedCenter = centerLineString.coordinateFromStart(distance: centerCoordDistance) {
+                    center = adjustedCenter
+                }
+                
+                followingMobileCamera.center = center
+                followingCarPlayCamera.center = center
             }
             
-            let averageIntersectionDistances = routeProgress.route.legs.map { (leg) -> [CLLocationDistance] in
-                return leg.steps.map { (step) -> CLLocationDistance in
-                    if let firstStepCoordinate = step.shape?.coordinates.first,
-                       let lastStepCoordinate = step.shape?.coordinates.last {
-                        let intersectionLocations = [firstStepCoordinate] + (step.intersections?.map({ $0.location }) ?? []) + [lastStepCoordinate]
-                        let intersectionDistances = intersectionLocations[1...].enumerated().map({ (index, intersection) -> CLLocationDistance in
-                            return intersection.distance(to: intersectionLocations[index])
-                        })
-                        let filteredIntersectionDistances = intersectionDistances.filter { $0 > 20 }
-                        let averageIntersectionDistance = filteredIntersectionDistances.reduce(0.0, +) / Double(filteredIntersectionDistances.count)
-                        return averageIntersectionDistance
-                    }
-                    
-                    return 0.0
-                }
+            if options.followingCameraOptions.zoomUpdatesAllowed {
+                followingMobileCamera.zoom = CGFloat(self.zoom(coordinatesToManeuver + coordinatesForManeuverFraming,
+                                                               pitch: pitch,
+                                                               edgeInsets: viewportPadding,
+                                                               defaultZoomLevel: 2.0,
+                                                               maxZoomLevel: 16.35))
```

Comment:
What about making `zoom` max/min configurable too? Especially that zoom level `2` seems too distant for "following" mode. Anyway, I would recommend extracting all the hardcoded values in constants, at least repeating ones.

Reply:
Yep, some of those will be extracted to match Android implementation.